nodejs/aws-infra/experimental/ec2/securityGroup.ts

```diff
@@ -37,6 +40,11 @@ export class SecurityGroup extends pulumi.ComponentResource {
         this.registerOutputs();
     }
 
+    /** @internal */
+    public static isSecurityGroupInstance(obj: any): obj is SecurityGroup {
+        return !!(<SecurityGroup>obj).__isSecurityGroupInstance;
+    }
```

Comment:
It's a shame these are needed.  I would have hoped we could just check for "is Input<string>" more easily - but I can see that probably isn't easy to implement.  The public API here feels simpler though - so I still do generally like this approach.

Reply:
👍